### PR TITLE
Fixes #28136 - add available_*(zones,flavors,networks) for GCE

### PR DESCRIPTION
**Pull request by kgaikwad**
+ adding @jyejare into loop

### CODE REVIEW

File: app/models/concerns/fog_extensions/google/server.rb [hunk 1]
```diff
@@ -7,8 +7,6 @@ module Server
       EXTERNAL_NAT_NAME = "External NAT".freeze
       EXTERNAL_NAT_TYPE = "ONE_TO_ONE_NAT".freeze
 
-      delegate :flavors, :to => :service
```
Comment (ezr-ondrej) [inline app/models/concerns/fog_extensions/google/server.rb:10]: Why this is no longer valid? :thinking:

**Review by ezr-ondrej — COMMENTED** [review]
Make sense, works well - I just don't understand the removal (probably just because I do not understand flavors vs machine-types). Could you please explain why we remove that line?

**Comment by kgaikwad** [comment]
> Make sense, works well - I just don't understand the removal (probably just because I do not understand flavors vs machine-types). Could you please explain why we remove that line?

@ezr-ondrej,
I don't think there is a difference between flavors vs machine-types except naming.
Previously, fog-google was using `flavor` model & `flavors` collection but later on those are removed and `machine_types` was added. On GCE side, it uses `machine_type` resource.

Reference links - 
https://github.com/fog/fog-google/commit/1b5f14fc76904e32fe1401648e73606b974240ae
https://github.com/fog/fog-google/commit/97e4ed3c5167bb410b7e4e0ef9a231b87b4a4aba 
https://github.com/fog/fog-google/commit/521a09be19c78ce106c4bad358ce5f257b989f8b

That's why I removed this line as it's like a dead code.

**Review by ezr-ondrej — APPROVED** [review]
Thanks for the clarification @kgaikwad!
And thanks for the addition :+1:

**Pull request merged**